pyrobosim/pyrobosim/core/robot.py

```diff
@@ -280,7 +282,10 @@ def plan_path(self, start=None, goal=None):
                 return None
             goal = goal_node.pose
 
-        path = self.path_planner.plan(start, goal)
+        if start.is_approx(goal,rel_tol=0.02, abs_tol=0.02):
+            path = Path([start, goal])
+        else:
+            path = self.path_planner.plan(start, goal)
```

Comment:
I don't think this is needed. Planners should be responsible for implementing the direct bypass from start to goal, which the RRT planner for sure does. It also violates the single-responsibility role of a motion planner.

Reply:
I'm seeing multiple occurrences of planning failures after arriving at a door and getting an open failure.  I was thinking that maybe it was due to something inside RRT related to resolution.  So long as we have strategy for consistent handling of this case I agree with your comments.